### DISCUSSION TITLE
<3>WSL (3217) ERROR: UtilTranslatePathList:2530: Failed to translate E:\

### DISCUSSION THREAD

**Discussion by llwwbb** [microsoft/WSL#8809] [Q&A] [answered]
<img width="549" alt="image" src="https://user-images.githubusercontent.com/7372183/189086996-ea0d1d64-5c05-43dd-98f4-58414da54cfc.png">
Since I unplugged a hard drive on the computer, this prompt appears every time I open wsl, what's wrong?

**Comment by NotTheDr01ds** [comment]
Have you rebooted since then or done a `wsl --shutdown`?

**Reply by llwwbb** [reply]
> Have you rebooted since then or done a `wsl --shutdown`?

yes I have. It still exists.

**Comment by Biswa96** [comment, marked as answer]
Try to remove that `E:\green_xyz` directory in PATH variable in Control Panel.

**Reply by llwwbb** [reply]
It works! Thank you

**Reply by Biswa96** [reply]
But the reason for this failure is a mystery now. Can you provide the full error message? What is the `E:\green_xyz` path, symlink or directory? What is the type of `E:\` drive? Is it compressed or encrypted? It may help WSL devs to fix the issue.

**Reply by llwwbb** [reply]
the path is E:\Green_QQGame_5.39.57734.0_41\Hall.57821, a directory. and E drive is just a normal drive, no compressed or encrypted.

**Reply by NotTheDr01ds** [reply]
@Biswa96 

> Try to remove that E:\green_xyz directory in PATH variable in Control Panel.

Good thought, by the way.

> But the reason for this failure is a mystery now.

I might be misunderstanding you, but it seems obvious based on the original question:

> Since I unplugged a hard drive on the computer, this prompt appears every time I open wsl

Since the `E:\` drive was removed, it didn't get auto-mounted by WSL, and so `/init` wasn't able to translate the `E:\Green_QQGame_5.39.57734.0_41\Hall.57821` into the corresponding `/mnt/e/Green_QQGame_5.39.57734.0_41/Hall.57821` to add it to the Linux `PATH`.

I'm able to reproduce it by adding `Z:\xyz` to my path (I don't have Z drive at the moment), starting a new Terminal window (so that it has the updated environment) and trying to launch any WSL distribution in it.

It's also seen in the `wslpath` command to some degree:

```
$ wslpath -u C:\\
/mnt/c
$ wslpath -u Z:\\
wslpath: z:\
# with error code
$ echo $?
1
```

I think I'd call it a "bug", since (IMHO) WSL should discard invalid paths.  But it's also a harmless message.  WSL still launches.  It probably should be a `WARNING` (or even `INFO`) rather than an `ERROR`, at least.
